src/components/toolbar/navigator-button.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useContext } from "react";
+import { observer } from "mobx-react";
+
+import { TileToolbarButton } from "./tile-toolbar-button";
+import { IToolbarButtonComponentProps } from "./toolbar-button-manager";
+
+import HideNavigatorIcon from "../../assets/icons/hide-navigator-icon.svg";
+import ShowNavigatorIcon from "../../assets/icons/show-navigator-icon.svg";
+import { TileModelContext } from "../tiles/tile-api";
+
+const isNavigatorSupported = (
+  model: any
+): model is { isNavigatorVisible: boolean; hideNavigator: () => void; showNavigator: () => void } => {
+  return (
+    model &&
+    typeof model.isNavigatorVisible === "boolean" &&
+    typeof model.hideNavigator === "function" &&
+    typeof model.showNavigator === "function"
+  );
+};
+
+export const NavigatorButton = observer(function NavigatorButton({ name }: IToolbarButtonComponentProps) {
```

Comment:
Since this is a shared component, a comment explaining what it is would be helpful.